packages/core/src/request-builder/header-builder/csrf-token-header.ts

```diff
@@ -13,45 +12,44 @@ const logger = createLogger({
 });
 
 /**
- * Get CSRF token and cookies for a request with destination related headers.
- * @param request The request to get CSRF headers for.
- * @param headers Destination related headers to include in the CSRF fetch request.
+ * Get CSRF token and cookies for a destination and request configuration. The CSRF token and cookies will be retrieved based on the url of the destination and custom configuration given by the `requestConfig`.
+ * @param destination The destination to get the headers from
+ * @param requestConfig An http request configuration containing additional information about the request, like url or headers
  * @returns A promise to an object containing the CSRF related headers
  */
-export async function buildCsrfHeaders<RequestT extends ODataRequestConfig>(
-  request: ODataRequest<RequestT>,
-  headers: MapType<string>
+export async function buildCsrfHeaders<T extends HttpRequestConfig>(
```

Comment:
This is a breaking change I guess? I would assume nobody uses this but to be safe we would have to overload the method to keep both both methods or you introduce a new method and deprecate the old one.

Reply:
Yes, this is a breaking change, but we did not release since the last change yet, therefore I would say, if someone uses canary, that's ok.

Reply:
(I was already aware of that when opening the old PR, but it was just too much for one PR.)

Reply:
ok